src/sqlparser/src/parser.rs

```diff
@@ -2578,14 +2579,24 @@ impl Parser<'_> {
         let include_options = self.parse_include_options()?;
 
         // PostgreSQL supports `WITH ( options )`, before `AS`
-        let with_options = self.parse_with_properties()?;
+        let mut with_options = self.parse_with_properties()?;
 
         let option = with_options
             .iter()
             .find(|&opt| opt.name.real_value() == UPSTREAM_SOURCE_KEY);
         let connector = option.map(|opt| opt.value.to_string());
-
-        let format_encode = if let Some(connector) = connector {
+        let contain_webhook = if let Some(connector) = &connector
+            && connector.contains("webhook")
+        {
+            // for webhook source, we clear all the options and use `webhook_info` to store the webhook info
+            with_options.clear();
```

Comment:
Why? Sounds error-prone 🥵

I think `webhook_info` is built from the `VALIDATE` clause, nothing to do with the `WITH` clause.

Reply:
Sorry, I though I've already replied this. If we keep the `with_options`, it will lead to a lot of modification in the validation code of `with_options`. What's worse, the `with_options` will be persisted in `CreateTable` metadata. It make thing more error-prone.

Reply:
> If we keep the `with_options`, it will lead to a lot of modification in the validation code of `with_options`

Can you show the "validation code" mentioned here?

Reply:
Complements: The `WITH` cluase actually also contains Hummock table options, although currently there is only one option `retention_seconds`.

Reply:
There are twofold of the "validation code":
1. In `parse_format_encode_with_connector`, fe will assign `FormatEncodeOption` for the source.
2. In `check_create_table_with_source`, fe will consider webhook source as a real source and return error for `FormatEncodeOptions`'s absence.

> Complements: The WITH cluase actually also contains Hummock table options, although currently there is only one option retention_seconds.

Got it. Let me find a way to avoid `with_options.clear();`

Reply:
Fixed, please take a look